src/main/java/gnu/io/RXTXPort.java

```diff
@@ -871,94 +904,54 @@ public void addEventListener(
 			throw new TooManyListenersException();
 		}
 		SPEventListener = lsnr;
-		if( !MonitorThreadAlive )
+		if (this.monThread == null)
 		{
-			MonitorThreadLock = true;
+			this.monitorThreadState.writeLock().lock();
 			monThread = new MonitorThread();
 			monThread.setName("RXTXPortMonitor("+name+")");
 			monThread.start();
-			waitForTheNativeCodeSilly();
-			MonitorThreadAlive=true;
+			try {
+				this.monitorThreadReady.await();
```

Comment:
@MrDOS Any error that occurs during `initialise_event_info_struct` causes the monitor thread to bail. As the thread dies without ever releasing the lock and pounding the signal, it indefinitely locks the main thread inside this `await()` call.

We've been testing this PR for a couple of weeks on a monolithic JVM with several webapps and decent network traffic, and keep encountering this hang once in a while. My theory is that due to an inrush burst of filesystem/network activity, the number of file descriptors temporarily exceeds `FD_SETSIZE`, causing the kernel to hand out `fd` numbers over that limit, which (rightly) causes `initialise_event_info_struct` to fail, locking up the whole JVM.